src/main/scala/firrtl/transforms/RemoveWires.scala

```diff
@@ -23,16 +23,20 @@ import scala.util.{Try, Success, Failure}
 class RemoveWires extends Transform with DependencyAPIMigration {
 
   override def prerequisites = firrtl.stage.Forms.MidForm ++
-    Seq( Dependency(passes.LowerTypes),
+    Seq( Dependency(passes.NewLowerTypes),
          Dependency(passes.Legalize),
+         Dependency(passes.ResolveKinds),
          Dependency(transforms.RemoveReset),
          Dependency[transforms.CheckCombLoops] )
 
   override def optionalPrerequisites = Seq(Dependency[checks.CheckResets])
 
   override def optionalPrerequisiteOf = Seq.empty
 
-  override def invalidates(a: Transform) = false
+  override def invalidates(a: Transform) = a match {
```

Comment:
Was this change necessary for this PR? Otherwise it should probably be separated into another PR.

Reply:
I did that while debugging. I can move it to a separate PR.

Reply:
Now a separate PR: #1797